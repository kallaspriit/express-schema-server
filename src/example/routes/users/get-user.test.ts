import * as HttpStatus from "http-status-codes";
import * as supertest from "supertest";

import setupApp from "../../app";

let app: supertest.SuperTest<supertest.Test>;

describe("get-user", () => {
  beforeEach(async () => {
    app = supertest(await setupApp());
  });

  it("should return registered user info", async () => {
    const createResponse = await app.post("/users").send({
      name: "Jack Daniels",
      email: "jack@daniels.com",
    });

    expect(createResponse.status).toEqual(HttpStatus.CREATED);
    expect(createResponse.body.success).toBe(true);
    expect(createResponse.body.payload).toMatchSnapshot();

    const getResponse = await app.get(`/users/${createResponse.body.payload.id}`).send();

    expect(getResponse.status).toEqual(HttpStatus.OK);
    expect(getResponse.body.success).toBe(true);
    expect(getResponse.body.payload).toMatchSnapshot();
  });

  it("should return validation error for invalid user id", async () => {
    const getResponse = await app.get("/users/abc").send();

    expect(getResponse.status).toEqual(HttpStatus.BAD_REQUEST);
    expect(getResponse.body.success).toBe(false);
    expect(getResponse.body.validationErrors).toMatchSnapshot();
  });

  it("should return 404 not found for non-existing user", async () => {
    const getResponse = await app.get("/users/666").send();

    expect(getResponse.status).toEqual(HttpStatus.NOT_FOUND);
    expect(getResponse.text).toMatchSnapshot();
  });
});
