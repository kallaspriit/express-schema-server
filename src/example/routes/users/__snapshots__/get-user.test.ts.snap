// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-user should return 404 not found for non-existing user 1`] = `"User with id \\"666\\" could not be found"`;

exports[`get-user should return registered user info 1`] = `
Object {
  "email": "jack@daniels.com",
  "id": 1,
  "name": "Jack Daniels",
}
`;

exports[`get-user should return registered user info 2`] = `
Object {
  "email": "jack@daniels.com",
  "id": 1,
  "name": "Jack Daniels",
}
`;

exports[`get-user should return validation error for invalid user id 1`] = `
Array [
  Object {
    "code": "INVALID_TYPE",
    "description": "User id",
    "message": "Expected type number but found type string",
    "params": Array [
      "number",
      "string",
    ],
    "path": "#/id",
  },
]
`;
